src/lib/Microsoft.Health.Extensions.Fhir.R4/Telemetry/Exceptions/FhirServiceExceptionProcessor.cs

```diff
@@ -0,0 +1,82 @@
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            var (customException, errorName) = CustomizeException(exception);
+
+            logger.LogError(customException);
+
+            string exceptionName = customException.Equals(exception) ? $"{ErrorType.FHIRServiceError}{errorName}" : customException.GetType().Name;
+            _exceptionTelemetryProcessor.LogExceptionMetric(customException, logger, FhirClientMetrics.HandledException(exceptionName, ErrorSeverity.Critical));
+        }
+
+        public static (Exception customException, string errorName) CustomizeException(Exception exception)
+        {
+            EnsureArg.IsNotNull(exception, nameof(exception));
+
+            string message;
+            string errorName;
+
+            switch (exception)
+            {
+                case FhirOperationException _:
+                    var status = ((FhirOperationException)exception).Status;
+                    switch (status)
+                    {
+                        case HttpStatusCode.Forbidden:
+                            message = FhirResources.FhirServiceAccessForbidden;
+                            string helpLink = "https://docs.microsoft.com/azure/healthcare-apis/iot/deploy-iot-connector-in-azure#accessing-the-iot-connector-from-the-fhir-service";
+                            errorName = nameof(FhirServiceErrorCode.AuthorizationError);
+                            return (new UnauthorizedAccessFhirServiceException(message, exception, helpLink, errorName), errorName);
+                        case HttpStatusCode.NotFound:
+                            message = FhirResources.FhirServiceNotFound;
+                            errorName = nameof(FhirServiceErrorCode.ConfigurationError);
+                            return (new InvalidFhirServiceException(message, exception, errorName), errorName);
+                        default:
+                            return (exception, status.ToString());
+                    }
+
+                case UriFormatException _:
+                    message = FhirResources.FhirServiceUriFormatInvalid;
+                    errorName = nameof(FhirServiceErrorCode.ConfigurationError);
+                    return (new InvalidFhirServiceException(message, exception, errorName), errorName);
+
+                case HttpRequestException _:
+                    message = FhirResources.FhirServiceHttpRequestError;
+                    errorName = nameof(FhirServiceErrorCode.ConfigurationError);
+                    return (new InvalidFhirServiceException(message, exception, errorName), errorName);
+
+                case MsalServiceException _:
+                    string errorCode = ((MsalServiceException)exception).ErrorCode;
+                    message = FhirResources.FhirServiceMsalServiceError;
+                    errorName = $"{nameof(FhirServiceErrorCode.ConfigurationError)}_{errorCode}";
```

Comment:
Should this be auth error instead?

Reply:
From my testing, no, because `MsalServiceException` was given for invalid FHIR service URLs such as https://clai1015.azurehealthcareapis.com/metadata, https://clai1015.azurehealthcareapis.com///, https://sdfsldfkwr389lksnldsf.com

In commit 9ab3837, I've scoped `MsalServiceException` down by checking the `ErrorCode` property.